biz.aQute.bndlib/src/aQute/bnd/build/Project.java

```diff
@@ -2112,6 +2112,15 @@ public void test(File reportDir, List<String> tests) throws Exception {
 			IO.delete(reportDir);
 			tester.setReportDir(reportDir);
 		}
+		if (tests == null) {
+			// check system properties: gradle
+			String singleTest = System.getProperty("integTest.single");
```

Comment:
First, this is for after 3.4. 

Second, we can modify the BndPlugin for gradle and bnd-testing-maven-plugin for maven to support a property which can be used to set the second argument to `Project.test(File reportDir, List<String> tests)`.